### ISSUE TITLE
When downloading a file, the file name is not suggested for saving

### ISSUE THREAD

**Issue by joao-paulo-parity** [Nheko-Reborn/nheko#688]
### The Problem

When I click to download a file:

![image](https://user-images.githubusercontent.com/77391175/130349338-a0ef02e2-627b-41cd-bbc2-5defce8e0da7.png)

The file name is not suggested:

![image](https://user-images.githubusercontent.com/77391175/130349478-0d84b0db-5321-42a5-a3ac-c37a9a916b56.png)

In the screenshot above, I would expect the file name (`2021-08-...`) to be suggested for the file when I am saving it, but the name is empty.

### The Solution

Suggest the file name when downloading a file.

### Happens in the latest version

- [X] Yes, this feature is still missing.

**Comment by deepbluev7** [comment]
This works on KDE, Linux. What version did experience this with, what platform, how did you install Nheko, etc? Really, this should just have been a bug report, because that has all the fields that I now have to ask for...

**Comment by deepbluev7** [comment]
Also that file looks weird, because it is size 0.

**Comment by deepbluev7** [comment]
Could you also check, if when you click on "View raw event" it has a filename or just a body field?

**Comment by joao-paulo-parity** [comment]
> this should just have been a bug report

I thought it was not implemented. That was my fault. Thanks for the heads-up.

> This works on KDE, Linux. What version did experience this with, what platform, how did you install Nheko, etc?

I'm running Arch Linux

```
> uname -a
Linux reaysawa 5.12.15-arch1-1 #1 SMP PREEMPT Wed, 07 Jul 2021 23:35:29 +0000 x86_64 GNU/Linux
```

It was installed through Flatpak and I'm using [a portal](https://github.com/flatpak/xdg-desktop-portal) for the file picker (maybe that's the problem? It works for the browsers I have installed through Flatpak, though).

```
> flatpak info io.github.NhekoReborn.Nheko

nheko - Desktop client for the Matrix protocol

          ID: io.github.NhekoReborn.Nheko
         Ref: app/io.github.NhekoReborn.Nheko/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 0.8.2
     License: GPL-3.0-or-later and CC-BY
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 46.4 MB
     Runtime: org.kde.Platform/x86_64/5.15
         Sdk: org.kde.Sdk/x86_64/5.15

      Commit: 1107d8f31a5392d783ab617249efe3a43d02e0e97fe913a026cfccfd5c6dd505
      Parent: fd6bc4192b34b4e2b4132f0150e3dd77e585ffeaa786dcda56b8cbd22d44cb3e
     Subject: Fix data to be date in appdata.xml (13cb9090)
        Date: 2021-04-29 09:57:55 +0000
```

> Could you also check, if when you click on "View raw event" it has a filename or just a body field?


```
{
    "content": {
        "body": "2021-08-20T12:45:40.053Z__2-log.txt",
        "info": {
            "mimetype": "",
            "size": 0
        },
        "msgtype": "m.file",
        "url": "mxc://server/AHCEkNGFkfKbDPtzGDRnffCs"
    },
    "event_id": "$azBOR0am3K5H_8XXGXFj3YoXm_onBrlNJxhABWggJdo",
    "origin_server_ts": 1629622584081,
    "sender": "@bot:server",
    "type": "m.room.message",
    "unsigned": {
        "age": 2208
    }
}
```

**Comment by deepbluev7** [comment]
Okay, so this could be a flatpak issue or that the message doesn't have a filename in the json. We may not be falling back to the body in those cases. I'll fix that.

**Comment by deepbluev7** [comment]
Hm, we are actually using body to set the filename, if filename is unset. So this might be an issue with the Qt flatpak portal. Does the file save dialog open in your download folder or somewhere else?

**Comment by joao-paulo-parity** [comment]
> Does the file save dialog open in your download folder or somewhere else?

It opens in my home folder

**Comment by deepbluev7** [comment]
So then the flatpak portal completely ignores the path we give it. Might be because of permissions or something else, will have to investigate. Is that the KDE or GTK portal?

**Comment by joao-paulo-parity** [comment]
> Is that the KDE or GTK portal?

It is GTK

```
> xprop
_NET_WM_DESKTOP(CARDINAL) = 2
WM_STATE(WM_STATE):
		window state: Normal
		icon window: 0x0
I3_FLOATING_WINDOW(CARDINAL) = 1
WM_TRANSIENT_FOR(WINDOW): window id # 0x3c00012
_NET_WM_STATE(ATOM) = _NET_WM_STATE_MODAL
WM_HINTS(WM_HINTS):
		Client accepts input or input focus: True
		Initial state is Normal State.
		window id # of group leader: 0x4800001
_GTK_THEME_VARIANT(UTF8_STRING) = "dark"
XdndAware(ATOM) = BITMAP
_NET_WM_OPAQUE_REGION(CARDINAL) = 0, 0, 1536, 431
WM_WINDOW_ROLE(STRING) = "GtkFileChooserDialog"
_NET_WM_WINDOW_TYPE(ATOM) = _NET_WM_WINDOW_TYPE_DIALOG
_NET_WM_SYNC_REQUEST_COUNTER(CARDINAL) = 75571550, 75571551
_NET_WM_USER_TIME(CARDINAL) = 29934710
_NET_WM_USER_TIME_WINDOW(WINDOW): window id # 0x481215d
WM_CLIENT_LEADER(WINDOW): window id # 0x4800001
_NET_WM_PID(CARDINAL) = 1445
WM_LOCALE_NAME(STRING) = "en_US.UTF-8"
WM_CLIENT_MACHINE(STRING) = "reaysawa"
WM_NORMAL_HINTS(WM_SIZE_HINTS):
		program specified minimum size: 715 by 431
		program specified base size: 0 by 0
		window gravity: NorthWest
WM_PROTOCOLS(ATOM): protocols  WM_DELETE_WINDOW, WM_TAKE_FOCUS, _NET_WM_PING, _NET_WM_SYNC_REQUEST
WM_CLASS(STRING) = "xdg-desktop-portal-gtk", "Xdg-desktop-portal-gtk"
WM_ICON_NAME(STRING) = "Save file"
_NET_WM_ICON_NAME(UTF8_STRING) = "Save file"
WM_NAME(STRING) = "Save file"
_NET_WM_NAME(UTF8_STRING) = "Save file"
```

> So then the flatpak portal completely ignores the path we give it. Might be because of permissions or something else, will have to investigate.

I tried `sudo flatpak override --filesystem=host io.github.NhekoReborn.Nheko` but it made no difference ([could have been the issue](https://github.com/flathub/io.github.antimicrox.antimicrox/pull/7/files)). It might be related to https://github.com/flatpak/xdg-desktop-portal-gtk/issues/191#issuecomment-474246486.

FWIW I can save the file perfectly file, it's just that the name doesn't get suggested. In any case, feel free to close this issue since the problem might not be here.

**Comment by deepbluev7** [comment]
We usually still try to figure out, what is wrong and if we can fix it, so this will stay open until we investigate further. But it is not a priority. Thanks for providing all the information so far!

**Comment by deepbluev7** [comment]
We can't fix it, it is a Qt bug: https://bugreports.qt.io/browse/QTBUG-91091

You can build Nheko yourself or use a distro package, if it is that annoying.

**Comment by deepbluev7** [comment]
Well, we can fix it, but we would need to work on the Qt code, I guess :D

**Comment by pizdjuk** [comment]
building a distro package works despite of QT-bug?

**Comment by deepbluev7** [comment]
Yes, this only affects flatpaks as far as I am aware or if you force nheko to use a portal.

**Comment by gabrc52** [comment]
I can confirm. Is there a way to bump the upstream Qt issue?

**Comment by deepbluev7** [comment]
You can upvote it in their issue tracker or pay them money to fix it.
